device/thunder_ripple_sdk/src/client/thunder_client2.rs

```diff
@@ -0,0 +1,157 @@
+use super::thunder_plugins_status_mgr::BrokerCallback;
+
+pub struct ThunderClientBuilder;
+
+pub type BrokerSubMap = HashMap<String, Vec<MpscSender<DeviceResponseMessage>>>;
+
+#[derive(Debug, Clone)]
+pub struct ThunderClient {
+    client: ThunderAsyncClient,
+    subscriptions: Arc<RwLock<BrokerSubMap>>,
+    callbacks: Arc<RwLock<HashMap<u64, Option<OneShotSender<DeviceResponseMessage>>>>>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct DefaultThunderResult {
+    pub success: bool,
+}
+
+impl ThunderClient {
+    fn add_to_callback(&self, request: &ThunderAsyncRequest) {}
+
+    // if already subscibed updated handlers
+    fn check_sub(&self, request: &DeviceSubscribeRequest) -> Option<ThunderAsyncRequest> {
+        None
+    }
+
+    // if only one handler cleanup
+    fn check_unsub(&self, request: &DeviceUnsubscribeRequest) -> Option<ThunderAsyncRequest> {
+        None
+    }
+}
+
+#[async_trait]
+impl DeviceOperator for ThunderClient {
+    async fn call(&self, request: DeviceCallRequest) -> DeviceResponseMessage {
+        let (tx, rx) = oneshot::channel::<DeviceResponseMessage>();
+        let async_request = ThunderAsyncRequest::new(DeviceChannelRequest::Call(request));
+        self.add_to_callback(&async_request);
+        self.client.send(async_request).await;
+        match rx.await {
+            Ok(response) => response,
+            Err(_) => DeviceResponseMessage {
+                message: Value::Null,
+                sub_id: None,
+            },
+        }
+    }
+
+    async fn subscribe(
+        &self,
+        request: DeviceSubscribeRequest,
+        handler: mpsc::Sender<DeviceResponseMessage>,
+    ) -> DeviceResponseMessage {
+        if let Some(subscribe_request) = self.check_sub(&request) {
+            let (tx, rx) = oneshot::channel::<DeviceResponseMessage>();
+            self.add_to_callback(&subscribe_request);
+            self.client.send(subscribe_request).await;
+            rx.await.unwrap()
+        } else {
+            DeviceResponseMessage {
+                message: Value::Null,
+                sub_id: None,
+            }
+        }
+    }
+
+    async fn unsubscribe(&self, request: DeviceUnsubscribeRequest) {
+        // deprecate
+    }
+}
+
+#[derive(Debug)]
+pub struct ThunderClientManager;
+
+impl ThunderClientManager {
+    fn manage(
+        client: ThunderClient,
+        mut request_tr: Receiver<ThunderAsyncRequest>,
+        mut response_tr: Receiver<ThunderAsyncResponse>,
+    ) {
+        let client_c = client.clone();
+        tokio::spawn(async move {
+            loop {
+                request_tr = client_c.client.start("", request_tr).await;
+                error!("Thunder disconnected so reconnecting")
+            }
+        });
+
+        tokio::spawn(async move {
+            while let Some(v) = response_tr.recv().await {
+                // check with thunder client callbacks and subscriptions
```

Comment:
@satlead : What's the purpose of `ThunderClientManager` here, is it supposed to receive requests and send them to a new thunder broker insteance, and then it also listens for responses from that broker and sends the response back to the requestor? Trying to understand what @nnaveen979 needs to do here.

Reply:
Thats a sample code he is supposed to delete this file